app/modules/http.c

```diff
@@ -7,224 +7,162 @@
 #include "lauxlib.h"
 #include "platform.h"
 #include "cpu_esp8266.h"
+#include "osapi.h"
 #include "httpclient.h"
 
 static int http_callback_registry  = LUA_NOREF;
 static lua_State * http_client_L = NULL;
 
-static void http_callback( char * response, int http_status, char * full_response )
+static void http_callback( char * response, int http_status, char * full_response, int content_length )
 {
+
 #if defined(HTTPCLIENT_DEBUG_ON)
-  c_printf( "http_status=%d\n", http_status );
-  if ( http_status != HTTP_STATUS_GENERIC_ERROR )
-  {
-    c_printf( "strlen(full_response)=%d\n", strlen( full_response ) );
-    c_printf( "response=%s<EOF>\n", response );
-  }
+	HTTPCLIENT_DEBUG( "http_status=%d\n", http_status );
```

Comment:
Looking through the code, there doesn't seem to be a consistent style (and none documented that I can find). I styled using leading tabs here because then at least the tab stops can be set within each person's editor.